src/main/java/com/google/sps/servlets/RecommenderServlet.java

```diff
@@ -36,20 +41,64 @@
 public class RecommenderServlet extends HttpServlet {
 
   private final String[] FACTORS = {"cost-rating", "crowd-rating"};
+  private DatastoreService datastore;
+
+  @Override
+  public void init(){
+    datastore = DatastoreServiceFactory.getDatastoreService();
+  }
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String name = "";
+    String category = "";
+    String description = "";
+    double lat = 0.0;
+    double lng = 0.0;
+    int costRating = 0;
+    int crowdRating = 0;
```

Comment:
There's no need to declare these variables here with the default values, please have the declaration where they are used (e.g. `String name = recommendation.getProperty(...)`).

This will also minimize their scope, see [`4.8.2.2 Declared when needed`](https://google.github.io/styleguide/javaguide.html#s4.8.2-variable-declarations)

Reply:
Good point, thanks!